db_stress_tool/no_batched_ops_stress.cc

```diff
@@ -2340,6 +2361,13 @@ class NonBatchedOpsStressTest : public StressTest {
     }
 
     ReadOptions ro(read_opts);
+    std::unique_ptr<ManagedSnapshot> snapshot = nullptr;
+    if (ro.auto_refresh_iterator_with_snapshot) {
+      snapshot = std::make_unique<ManagedSnapshot>(db_);
+      ro.snapshot = snapshot->snapshot();
```

Comment:
This seems to be a problem because we are acquiring a snapshot before getting `pre_read_expected_values` which makes the timing of the read effectively before reading `pre_read_expected_values` rather than between reading `pre_read_expected_values` and `post_read_expected_values`.

I'm sure other instances of creating snapshots here are similarly problematic.